### ISSUE TITLE
Search: -ordfav:user is broken

### ISSUE THREAD

**Issue by evazion** [danbooru/danbooru#4918]
https://danbooru.donmai.us/posts?tags=-ordfav:evazion

```
ERROR:  missing FROM-clause entry for table "favorites" at character 36
STATEMENT:  SELECT COUNT(*) FROM "posts" WHERE "favorites"."user_id" != $1
```

**Comment by sreeram-venkitesh** [comment]
Hi! Fredgido linked this issue to me in discord. Can I try working on this?

**Comment by Thayol** [comment]
Could this be solved by replacing `-ordfav:` with `-fav:`? The latter works as intended, and as far as I know there is no difference between the two.

**Comment by Thayol** [comment]
What does a negated `order` statement mean?

We have multiple types of `order`:
* `order:rank` has a limited result, so we could remove that from the results if negated.
* `order:landscape`, `order:score`, etc. return ALL posts matching the other tags, just in different order. Should the negated result in a `Post.none`?

**Comment by BrokenEagle** [comment]
> What does a negated order statement mean?

It's technically the same thing as `fav:` , except it presents posts in the order that they are in the favorite group. So it should operate exactly the same as `-fav:` .

There's also `ordfavgroup:` and `ordpool:` , which operate the same as `favgroup:` and `pool:` , though presenting posts as they appear in the favorite group and pool respectively.

**Comment by Thayol** [comment]
Yes, I made a PR for fixing `ordfav`, `ordpool`, and `ordfavgroup` before my previous comment.
Let me rephrase:

What does a `-order:score` or a `-order:comment` mean? If it means nothing, we could return an empty result instead of the error page.

Also, `order:score` is still an odd one in my mind because it doesn't purely "order" things but it has a limit, so I believe a `-order:score` might be valid in some cases.
